### DISCUSSION TITLE
a syntax tree visualization tool

### DISCUSSION THREAD

**Discussion by shi-yan** [tree-sitter/tree-sitter#1468] [Libraries & tools]
I need to work on tree-sitter syntax tree pruning and simplification,

I found the s-expression output hard to comprehend. I tried converting the tree into a json format, it's not any better.

I hence built this visualization tool for tree-sitter.

 in case you have a similar need, see

https://github.com/shi-yan/tree-sitter-viewer

![image1](https://github.com/shi-yan/tree-sitter-viewer/raw/master/screenshot1.png) 

![image2](https://github.com/shi-yan/tree-sitter-viewer/raw/master/screenshot2.png)

**Comment by alemuller** [comment]
That's pretty cool!

I can imagine someone writing a [kitten](https://sw.kovidgoyal.net/kitty/kittens_intro/#kittens) to display the AST, and controlling the view from nvim using [MessagePack](https://msgpack.org/).

**Comment by razzeee** [comment]
So that's for the actual grammar input and not what you're parsing? I guess your aware of `tree-sitter parse ./examples/basic.elm -D` then?

**Reply by shi-yan** [reply]
Thanks! I have never used the cli. I wasn't aware of -D (--debug-graph) , 

only after checking https://github.com/tree-sitter/tree-sitter/blob/master/cli/src/parse.rs I now know what it is.

the document doesn't mention anything http://tree-sitter.github.io/tree-sitter/creating-parsers
